demos/shapes-custom-question-authoring-demo/src/question/index.js

```diff
@@ -0,0 +1,339 @@
+                 */
+
+                if(init.response) {
+                    const shapeButtons = Array.from(this.el.querySelectorAll('.shape-wrapper button'));
+                    shapeButtons.forEach((shapeButton) => {
+                        if(shapeButton.getAttribute('data-index') === init.response) {
+                            shapeButton.classList.add('selected');
+                        }
+                    })
+                }
+            }
+
+            if (init.state === 'review') {
+                /**
+                 * If you want your custom question to support review state
+                 * (For a learner or instructor to be able to view their results in a read-only mode after having completed and submitted the assessment),
+                 * then make make sure to update your question's UI to display the previously submitted response Questions API has stored in the back end.
+                 * NOTE: this is required if you intend to use Reports API with your custom question (for example the session-detail-by-item report).
+                 */
+                if(init.response) {
+                    const shapeButtons = Array.from(this.el.querySelectorAll('.shape-wrapper button'));
+                    shapeButtons.forEach((shapeButton) => {
+                        if(shapeButton.getAttribute('data-index') === init.response) {
+                            shapeButton.classList.add('selected');
+                        }
+                    })
+                }
+                init.getFacade().disable();
+            }
+
+            
+
+            init.events.trigger('ready');
+        });
+    }
+
+    render() {
+        const { el, init, lrnUtils } = this;
+        const { question, response } = init;
+        
+        // TODO: Requires implementation
+        el.innerHTML = `
+            <div class="${PREFIX} lrn-response-validation-wrapper">
+                <div class="lrn_response_input">
+                    <div class="shapes flex-row"></div>
+                </div>            
+                <div class="${PREFIX}-checkAnswer-wrapper"></div>
+                <div class="${PREFIX}-suggestedAnswers-wrapper"></div>
+            </div>
+        `;
+
+        // Optional - Render optional Learnosity components like Check Answer Button, Suggested Answers List
+        // first before rendering your question's components
+        return Promise.all([
+            lrnUtils.renderComponent('SuggestedAnswersList', el.querySelector(`.${PREFIX}-suggestedAnswers-wrapper`)),
+            lrnUtils.renderComponent('CheckAnswerButton', el.querySelector(`.${PREFIX}-checkAnswer-wrapper`))
+        ]).then(([suggestedAnswersList]) => {
+            this.suggestedAnswersList = suggestedAnswersList;
+            
+            question.shapes && question.shapes.forEach((shape, index) => {
```

Comment:
We can simplify the null safe here since we already handling from the top with the fail safe message
```suggestion
            question.shapes.forEach((shape, index) => {
```